### ISSUE TITLE
about balanced_batch

### ISSUE THREAD

**Issue by Martiantian** [nanoporetech/remora#30]
Hello, thank you very much for providing tools for detecting RNA modifications. I‘m doing some testing with this tool and getting some good results. However, when I trained the model on a sample with a small proportion of modifications, I found that this parameter “--balanced-batch” was not sampling the training set to get the same number of positive and negative samples as I imagined, but the same number of positive and negative samples in the validation set. The combination of the code has confirmed my conjecture. I do not know much about machine learning. May I ask if this design can help the training of the model?

![image](https://user-images.githubusercontent.com/35797239/188606064-d047a837-b672-4c99-bdfd-ac38d45b7c00.png)
![image](https://user-images.githubusercontent.com/35797239/188606077-9fd76159-9e46-43ca-80e1-a09d68d4a20f.png)

**Comment by budnikm** [comment]
Hello,
To get the full picture you should look at the `__next__` function just above in the same class. There when `--balanced-batch` is used we randomly select the same number of examples for each class so that the resulting training batch is balanced. That way we can use all the examples in the dataset while still ensure that the model sees a balanced representation of the data. So this approach might help if you have highly unbalanced classes, like in your case.

**Comment by Martiantian** [comment]
Thank you very much for your prompt reply. I missed the part of the code you mentioned. If it has been handled in the way you described, then I have no other questions. Thanks again!
